### PR TITLE
mrjar: Set `@Target({..., MODULE})` in the root, and move `module-info` out.

### PR DESCRIPTION
**Pull request by cpovirk**
Fixes https://github.com/jspecify/jspecify/issues/245. But introduces a compile warning for users targeting old versions of Java, and introduces [runtime problems](https://bugs.openjdk.java.net/browse/JDK-8247797) if a user tries to read _the annotations on `@NullMarked` itself_.

Fixes https://github.com/jspecify/jspecify/issues/172. I didn't really need to do this in the same commit. I just happened to get things working that way first, and we've talked about at least trying this change, so here goes....

It's clear that there's no great option for `MODULE`. But given that we've already encountered problems in the wild with our old approach, let's try this new approach instead.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,7 +3,6 @@ plugins {
     id 'io.github.gradle-nexus.publish-plugin' version '1.1.0' apply false
     id 'com.diffplug.spotless' version '6.1.0'
     id 'net.ltgt.errorprone' version '2.0.2'
-    id 'org.javamodularity.moduleplugin' version '1.8.9'
```
Comment (KengoTODA) [inline build.gradle:6]: :+1: 
Better to stop depending on this plugin, it seems their development is not so active recently. I sent a PR two months ago and there is still no reaction 😂

File: gradle/mrjar.gradle [hunk 2]
```diff
@@ -27,14 +33,16 @@ tasks.named('compileJava9Java', JavaCompile).configure {
...
 jar {
     into('META-INF/versions/9') {
-        from sourceSets.java9.output
+        from(sourceSets.java9.output) {
+            /*
+             * As discussed above, we compiled all the classes again. But all
+             * that we want to include under META-INF/versions/9 is the
+             * module-info.
+             */
+            include 'module-info.class'
```
Comment (KengoTODA) [inline gradle/mrjar.gradle:42]: :+1:
Great comment, it will help us to understand the intention of this build script.

**Review by KengoTODA — APPROVED** [review]
LGTM, I've learned a lot from related discussions.
Also thank you making the build script more easy-to-understand!

File: build.gradle (hunk 1 above)
Comment (cpovirk) [reply build.gradle:6]: Thanks, that is good to know :)

**Pull request merged**